### DISCUSSION TITLE
How to save a JPG file like done with System.Drawing namespace?

### DISCUSSION THREAD

**Discussion by ilCosmico** [SixLabors/ImageSharp#2331] [Q&A] [answered]
I cannot figure out how to save a byte[] representing an image to JPG format like done with System.Drawing namespace.
Listed below is a code sample reproducing the issue.

```
using System.IO;
using System.Security.Cryptography;
using System.Drawing;
using SixLabors.ImageSharp;

[TestMethod, NUnit.Framework.Test, TestCategory("ImageSharp")]
public void SaveBytesToJpg()
{
    string fileName = "ChessTexture.jpg";
    var imageBytes = File.ReadAllBytes(fileName); 

    // System.Drawing
    using (MemoryStream ms = new MemoryStream(imageBytes))
    {
        var image1 = System.Drawing.Image.FromStream(ms);
        image1.Save("image1.jpg", System.Drawing.Imaging.ImageFormat.Jpeg);
    }

    // ImageSharp
    using (var image2 = SixLabors.ImageSharp.Image.Load(imageBytes))
    {
        image2.SaveAsJpeg("image2.jpg");
    }

    byte[] hash0 = GetHash(fileName);
    byte[] hash1 = GetHash("image1.jpg");
    byte[] hash2 = GetHash("image2.jpg");

    CollectionAssert.AreEqual(hash0, hash1); // OK
    CollectionAssert.AreEqual(hash1, hash2); // FAILED!

}

private byte[] GetHash(string fileName)
{
    using (SHA256 sha256 = SHA256.Create())
    {
        using (FileStream stream = File.OpenRead(fileName))
        {
            return sha256.ComputeHash(stream);
        }
    }
}
```

**Comment by JimBobSquarePants** [comment, marked as answer]
You can’t compare the output of encoders like that especially lossy ones. Minute differences in color transformation can lead to differences and the order of segments within an image is not guaranteed as the specification does not require many to be so. Plus you have different quality defaults and metadata handling strategies.

**Reply by ilCosmico** [reply]
@JimBobSquarePants so the only way I can compare them is by using a visual comparison among stored images?
I mean something like [this](https://rsmbl.github.io/Resemble.js/).

**Reply by antonfirsov** [reply]
Yes, using visual diff tools is a good approach. On larger scale, you can do programmatic pixel-by-pixel error accumulation, like we do in our test test suite, see #1129.
